content/concepts/did-ddo.md

```diff
@@ -369,7 +372,100 @@ Example:
   ]
 }
 ```
+ 
+#### User Input
+
+Sometimes, you may need some input before consuming a dataset or running an algorithm.
+Examples:
+
+- You want to know the desired sampling interval of data in your dataset, before the user is going to download it. Your dataset URL is 'www.example.com/mydata'. So you will define a field called "sampling", ask the user to enter a value and then this parameter is going to be added to the URL of your dataset (so the actual url will become 'www.example.com/mydata?sampling=10'
+- Before running an algorithm, you need to know how many iterations should the algo perform.  You are going to define a field called 'iterations', ask the user to enter a value and this parameter is stored in a specific location in your C2D pod, so algo can read it and use that value.
+
+Object consists of an array , in which each element defines a user required field.
+An element looks like:
+
+```json
+   {
+        "name":"surname",
```

Comment:
it's trivial but i don't think the surname/Name example is good here (for me  it was confusing for a moment) . I think it would be better to use on of the fields from the examples above (iterations/sampling)